### PR TITLE
132 cypress test for contact us form

### LINKED ISSUE #132: [Developer Experience]: Add Cypress tests for 'Contact Us' form
## Developer Experience Issue

**Please describe your Developer Experience Issue.**
A clear and concise description of what the problem is. Ex. I have an issue when [...]

Add Cypress tests for 'Contact Us' form.

**Describe the solution you'd like**
A clear and concise description of what you want to happen. Add any considered drawbacks.

Add a Cypress Test suite that would:

- Test if form would throw an error if not all inputs are filled in
- Test if form will submit properly when all form elements are filled in
- Test if an error message comes up under the appropriate form input if a change needs to be made

**Describe alternatives you've considered**

n/a

**Teachability, Documentation, Adoption, Migration Strategy**
If you can, explain how users will be able to use this and possibly write out a version in the docs.
Maybe a screenshot or design?

n/a

### PR DESCRIPTION
**Pull request by ghost**
## Description

Edited the current contact-form.tsx file to add some helper attributes to elements and added the test suite for the isolated component using cypress.

- Added 'data-cy' attributes for the name, email and message input fields and error span elements  in "../src/components/molecules/contact-form/contact-form.tsx"  file  in order to safely fetch them in the cypress tests
- Added test suite for contact-form component handling three cases: 
-- No name, email and message is provided
--  All three proper inputs are provided.
-- Only valid email is provided 

## Related Issue

Resolves: #132  

## Motivation and Context
n/a

## Screenshots (if appropriate):
Case 1: 
Screenshot 2023-05-25 at 4 54 48 PM

Case 2: 
Screenshot 2023-05-25 at 4 55 34 PM

Case 3:
Screenshot 2023-05-25 at 4 56 02 PM

### CODE REVIEW

**Review by chadstewart — CHANGES_REQUESTED** [review]
Thanks so much for your contribution. So just wanted to request some changes.

Outside of what I added, could you test the input fields separately please? Also could you add tests where you test where the input fields have have invalid characters or is in an invalid format and not just empty please?

Thanks much and looking forward to your update!

File: cypress/components/contactuscomponent.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      cy.get('@nameInputTextField').type('Tausif Qureshi')
+      cy.get('@emailInputTextField').type('tausif2606@gmail.com')
+      cy.get('@messageEmailTextArea').type('Hello tech is hiring!')
+
+      cy.intercept('POST','/api/process-email', (req) => {
+        const statusCode = 200
+        req.reply(statusCode, {data:"Email sent"})
+
+      })
+      cy.get('@sendMessageBtn').click()
```
[the remark this answers is not in the document]
Comment (Tausifqureshi786) [reply cypress/components/contactuscomponent.cy.tsx:45]: Screenshot 2023-05-30 at 6 46 53 PM

After running `npx cypress open --component` then in the browser test environment the above request made is shown to not be going to the actual server rather than is going to the stubbed request with a mock reply as shown in the above image. No emails are being sent to the given mail. I checked the [cypress docs](https://docs.cypress.io/guides/guides/network-requests#Command-Log) and it refers that a request that goes to the actual server is shown with a filled circle and the mocked requests endpoints are shown with the outlined circle as it is in the image.

[the remark this answers is not in the document]
Comment (chadstewart) [reply cypress/components/contactuscomponent.cy.tsx:45]: Ah okay, that makes sense. My bad for taking so long to get back to you.

[the remark this answers is not in the document]
Comment (chadstewart) [reply cypress/components/contactuscomponent.cy.tsx:45]: Could you work on the other issues that I pointed out in the review please?

**Comment by chadstewart** [comment]
Hey @Tausifqureshi786, just wanted to reach out and ask if you were able to make any further progress on this PR?

**Comment by dgodongm** [comment]
@chadstewart @Tausifqureshi786 Let me know if you'd like me to finish this up. In terms of consistency with https://github.com/TechIsHiring/techishiring-website/blob/dev/src/components/organisms/footer/footer.cy.tsx, should the test file here be moved to be in same folder as the component?
